server/src/main/java/org/candlepin/controller/CandlepinPoolManager.java

```diff
@@ -2268,7 +2268,8 @@ public void deletePools(Collection<Pool> pools, Collection<String> alreadyDelete
             // Revoke/delete entitlements
             if (!entitlements.isEmpty()) {
                 log.info("Revoking {} entitlements...", entitlements.size());
-                this.entitlementCurator.batchDelete(entitlements);
+                this.entitlementCertificateCurator.deleteByEntitlementIds(entitlementIds);
+                this.entitlementCurator.batchDeleteByIds(entitlementIds);
```

Comment:
See the comment in #2429 for more details

Has there been any additional testing around skipping EntitlementCurator's extra behavior on delete?

Reply:
The extra runtime consistency code was added back.